### DISCUSSION TITLE
Extensions development: Column "element" in the "#__extensions" table

### DISCUSSION THREAD

**Discussion by vinespie** [joomla/joomla-cms#43653] [Q&A] [answered]
Hello everyone, 

What is the correct formatting for the content of the "element" column in the "#__extensions" table?
Is there a recommendation?

![image](https://github.com/joomla/joomla-cms/assets/5129879/38278aca-b275-4b5e-a907-99bd54d5f2b2)

If type = 'component' => element start width 'com_'
If type = 'module' => element start width 'mod_'
If type = 'library' => element start width 'library_' ???
If type = 'plugin' => element start width 'plg_' ???

etc

Thank you for the responses.

**Comment by richard67** [comment, marked as answer]
There is no strict rule for that. The main rule is that the combination of type, element, folder and client_id has to be unique. The folder only relevant for extensions type "plugin", for others it is empty, and the client_id is only relevant for extensions of type "module".

**Reply by brianteeman** [reply]
> and the client_id is only relevant for extensions of type "module".

and templates and language files

**Comment by vinespie** [comment]
That's exactly what I thought; there are no rules for this column determined by the extension manifest file name. Thank you for your responses.
